force-app/main/default/classes/WeeklyIterator_TEST.cls

```diff
@@ -0,0 +1,101 @@
+        // every other Monday, 5 times or Oct 8 2099
+        RecurrenceRule rRule = new RecurrenceRule()
+            .withRuleString(
+                'FREQ=WEEKLY;COUNT=5;INTERVAL=2;BYDAY=MO;UNTIL=20991008T100000Z'
+            );
+        List<DateTime> expected = new List<DateTime>{
+            DateTime.newInstance(2020, 9, 14, 14, 0, 0),
+            DateTime.newInstance(2020, 9, 28, 14, 0, 0),
+            DateTime.newInstance(2020, 10, 12, 14, 0, 0),
+            DateTime.newInstance(2020, 10, 26, 14, 0, 0),
+            DateTime.newInstance(2020, 11, 9, 14, 0, 0) // crosses DST
+        };
+        WeeklyIterator iterator = new WeeklyIterator(rRule);
+
+        Test.startTest();
+        List<DateTime> actual = iterator.iterate(dateTimeStart);
+        Test.stopTest();
+
+        System.assertEquals(
+            expected,
+            actual,
+            'List of Dates returned from the iterator was not as expected.'
+        );
+    }
+
+    @IsTest
+    private static void shouldIterateEveryMondayAndTuesdayUntilGivenDate() {
+        // Start on Sept 1 2020, 2PM
+        DateTime dateTimeStart = DateTime.newInstance(2020, 9, 1, 14, 0, 0);
+        // every Monday and Tuesday until October 8 or 30 times
+        RecurrenceRule rRule = new RecurrenceRule()
+            .withRuleString('FREQ=WEEKLY;UNTIL=20201008T120000Z;BYDAY=MO,TU;COUNT=30');
+        List<DateTime> expected = new List<DateTime>{
+            DateTime.newInstance(2020, 9, 1, 14, 0, 0),
+            DateTime.newInstance(2020, 9, 7, 14, 0, 0),
+            DateTime.newInstance(2020, 9, 8, 14, 0, 0),
+            DateTime.newInstance(2020, 9, 14, 14, 0, 0),
+            DateTime.newInstance(2020, 9, 15, 14, 0, 0),
+            DateTime.newInstance(2020, 9, 21, 14, 0, 0),
+            DateTime.newInstance(2020, 9, 22, 14, 0, 0),
+            DateTime.newInstance(2020, 9, 28, 14, 0, 0),
+            DateTime.newInstance(2020, 9, 29, 14, 0, 0),
+            DateTime.newInstance(2020, 10, 5, 14, 0, 0),
+            DateTime.newInstance(2020, 10, 6, 14, 0, 0)
+        };
+        WeeklyIterator iterator = new WeeklyIterator(rRule);
+
+        Test.startTest();
+        List<DateTime> actual = iterator.iterate(dateTimeStart);
+        Test.stopTest();
+
+        System.assertEquals(
+            expected,
+            actual,
+            'List of Dates returned from the iterator was not as expected.'
+        );
+    }
+
+    @IsTest
+    private static void shouldIterateEveryThirdWeekendFor6Instances() {
+        // Start on Oct 12 2020, 8PM
+        DateTime dateTimeStart = DateTime.newInstance(2020, 10, 12, 20, 0, 0);
+        // every 3 weeks on Saturday, Sunday for 6 times
+        RecurrenceRule rRule = new RecurrenceRule()
+            .withRuleString('FREQ=WEEKLY;COUNT=6;INTERVAL=3;BYDAY=SA,SU');
+        List<DateTime> expected = new List<DateTime>{
+            DateTime.newInstance(2020, 10, 17, 20, 0, 0),
```

Comment:
This test is interesting and something we might want to think about in regards to the UI. Because technically they are saying the startDateTime is the first session, however here we do not return it as part of the results.

Reply:
I think of this test as outside of the scope of our UI, but within the scope of our generator in the spirit of making it reusable. I still definitely think that our UI can be more guided.